### ISSUE TITLE
[HOLD for payment 2023-02-15] [$2000] not able to select text of keyboard shortcuts when you open popup second time

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#13968]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:

1. open app
2. go to settings > About > View Keyboard shortcuts
3. open  the popup an close it
4. again open the popup and try to select text

## Expected Result:
not able to select text when you open popup second time

## Actual Result:
user should be able to select text

## Workaround:
unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?

- [ ] MacOS / Chrome 

**Version Number:** 1.2.47-0
**Reproducible in staging?:** y 
**Reproducible in production?:** y
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** 

https://user-images.githubusercontent.com/43996225/210454215-e770a2da-afdf-4dbe-ab23-2b6b7e45c4bb.mov

https://user-images.githubusercontent.com/43996225/210454442-330d11d9-8dde-4e24-bdcd-ec5a5c6d5a30.mp4

**Expensify/Expensify Issue URL:**
**Issue reported by:** @gadhiyamanan 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1672381522931449

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~0142c09828b956ad3b</li>
        <li>Upwork Job ID: 1613255747122876416</li>
        <li>Last Price Increase: 2023-02-01</li>
    </ul>
</details>

**Comment by jasperhuangg** [comment]
Hey! Don't really think this needs to be fixed, @joekaufmanexpensify do you agree?

**Comment by sketchydroide** [comment]
this is a rather specific BUG, but one nonetheless, and acording to WAQ, if it't reproducable it's a Bug...
Also I can see reasons a user would like to copy paste keyboard shortcuts, namely to send it to another user so 🤷🏼 

If you don't take it I will @jasperhuangg

**Comment by joekaufmanexpensify** [comment]
Yes, agreed. My understanding is if it's reproducible, we will fix it.

**Comment by joekaufmanexpensify** [comment]
Triaging:

From this SO: https://stackoverflow.com/c/expensify/questions/14418
- [ ] **The "bug" is actually a bug**: 
- [ ] **The bug is not a duplicate report of an existing GH (close the GH and add any novel details to the original GH instead)**: 
- [ ] **The bug is reproducible, following the reproduction steps:**
- [ ] **If you're unable to reproduce the bug, add the _Needs reproduction_ label. Comment on the issue outlining the steps you took to try to reproduce the bug, your results and tag the issue reporter and the Applause QA member who created the issue. Ask them to clarify reproduction steps and/or to check the reproduction steps again. Close issue.**: 
- [ ] **The GH template is filled out as fully as possible -- this means the GH body and title are clear (ie. could an external contributor understand it and work on it?):** 
- [ ] **The GH was created by an Expensify employee or a QA tester:** 
- [ ] **If the bug is an OldDot issue, you should decide whether it is widespread enough to justify fixing or it is better to wait for reunification. If it's better to wait, close the GH & provide this justification:** 
- [ ] **If there's a link to Slack, check the discussion to see if we decided not to fix it:** 
- [ ] **Decide if the GH should be resolved by an External contributor or Internal engineer, add the appropriate label:**

**Comment by joekaufmanexpensify** [comment]
I can reproduce the fact that the text is not selectable on the second try here. Whether we want the text to be selectable, I am less sure. I am going to bring to Slack. 

![2023-01-04_15-03-59 (1)](https://user-images.githubusercontent.com/54866469/210640745-b105ae0a-84c9-42a9-a565-8cdf750203ab.gif)

**Comment by joekaufmanexpensify** [comment]
Posted about this [here. ](https://expensify.slack.com/archives/C01SKUP7QR0/p1672863653244019)

**Comment by joekaufmanexpensify** [comment]
Bumped slack discussion.

**Comment by joekaufmanexpensify** [comment]
Still waiting for more discussion in Slack.

**Comment by joekaufmanexpensify** [comment]
Not overdue.

**Comment by jasperhuangg** [comment]
Hey sorry @sketchydroide I missed your message, feel free to take this on!

**Comment by joekaufmanexpensify** [comment]
Bumped slack thread. I think we should get crisp on what the expected behavior is here before we fix this. Either way, I think this is a bug, but just want to make sure we know whether this text should be copiable in all situations, or none (thinking the former).

**Comment by joekaufmanexpensify** [comment]
waiting for confirmation

**Comment by joekaufmanexpensify** [comment]
Chatted with Andre, and making this external.

**Comment by allroundexperts** [comment]
## Proposal

The modal when closed results in the body being focused. When the modal re-opens, body remains focused. 
To fix this, we need to remove the focus from body when modal opens.

Following needs to be changed:
diff```
diff --git a/src/components/Modal/index.web.js b/src/components/Modal/index.web.js
index 3cb6f3561..292abab40 100644
--- a/src/components/Modal/index.web.js
+++ b/src/components/Modal/index.web.js
@@ -23,6 +23,7 @@ const Modal = (props) => {
     const showModal = () => {
         setStatusBarColor(StyleUtils.getThemeBackgroundColor());
         props.onModalShow();
+        document.body.removeAttribute('tabindex');
     };
 
     return (
```

**Comment by syedsaroshfarrukhdot** [comment]
Proposal :-

To avoid any regression on modals where we don't need focus we should set focusable using props such as data pickers modal etc.

``` diff
diff --git a/src/components/KeyboardShortcutsModal.js b/src/components/KeyboardShortcutsModal.js
index c45ced82f..3fc9df81c 100644
--- a/src/components/KeyboardShortcutsModal.js
+++ b/src/components/KeyboardShortcutsModal.js
@@ -81,6 +81,7 @@ class KeyboardShortcutsModal extends React.Component {
                 type={modalType}
                 containerStyle={{...styles.keyboardShortcutModalContainer, ...StyleUtils.getKeyboardShortcutsModalWidth(this.props.isSmallScreenWidth)}}
                 onClose={KeyboardShortcutsActions.hideKeyboardShortcutModal}
+                focusable
             >
                 <HeaderWithCloseButton title={this.props.translate('keyboardShortcutModal.title')} onCloseButtonPress={KeyboardShortcutsActions.hideKeyboardShortcutModal} />
                 <View style={[styles.p5, styles.pt0]}>
diff --git a/src/components/Modal/BaseModal.js b/src/components/Modal/BaseModal.js
index 323df19c7..90f8db626 100644
--- a/src/components/Modal/BaseModal.js
+++ b/src/components/Modal/BaseModal.js
@@ -34,7 +34,6 @@ class BaseModal extends PureComponent {
         if (prevProps.isVisible === this.props.isVisible) {
             return;
         }
-
         Modal.willAlertModalBecomeVisible(this.props.isVisible);
     }
 
@@ -116,6 +115,7 @@ class BaseModal extends PureComponent {
                 animationInTiming={this.props.animationInTiming}
                 animationOutTiming={this.props.animationOutTiming}
                 statusBarTranslucent={this.props.statusBarTranslucent}
+                focusable={this.props.focusable}
             >
                 <SafeAreaInsetsContext.Consumer>
                     {(insets) => {
```

**Comment by getusha** [comment]
@aimane-chnaif there is no regression as of my tests you can confirm that by adding 
```javascript
                onBlur={()=>{
                    console.log("Modal Blurred")
                }}
```
the blur happens every time we close the modal

**Comment by syedsaroshfarrukhdot** [comment]
@aimane-chnaif Such As In Send Attachment Modal Text In Not Selectable From Default.

You can see in video below.

https://user-images.githubusercontent.com/81307212/212011692-215e7a8f-e925-4c60-af13-007015e634a5.mov

After Adding Focusable It Make Text Selectable In The Send Attachment Modal Where We Don't Need It So It Is A Regression I Believe.

https://user-images.githubusercontent.com/81307212/212011954-a4a9b1a7-acf5-4cb8-afeb-e06b5956ae56.mov

I proposed solution to add focusable only in KeyboardShortcutModal only because the scope of this issue was to make text selectable always only on this modal.

If you want I can identify any other instances if there is need to add focus on any other modals where are we facing text selectable issues where we need it.

**Comment by getusha** [comment]
this issue is if the cursor is I beam the text should be selectable 
ref: https://expensify.slack.com/archives/C049HHMV9SM/p1672770003595509?thread_ts=1672381522.931449&cid=C049HHMV9SM

**Comment by syedsaroshfarrukhdot** [comment]
> https://expensify.slack.com/archives/C049HHMV9SM/p1672770003595509?thread_ts=1672381522.931449&cid=C049HHMV9SM

I don't agree with it as above in my comment you can see reference is WhatsApp Keyboard Shotcut Modal where cursor is I-Beam but they have disable selection on the keyboard shortcut modal. So it is totally up to the team how to handle text selection.

**Comment by getusha** [comment]
https://user-images.githubusercontent.com/75031127/212069729-38b7de17-ba22-4d71-acca-a5a513739d30.mov

The text is not blocked by default actually, it's an issue which should be addressed and my [proposal](https://github.com/Expensify/App/issues/13968#issuecomment-1379445596) it fixes it all

**Comment by getusha** [comment]
And IMO since the proposal is fixing the issue, it's another thing that we want to apply it on a specific component or not

**Comment by s77rt** [comment]
### Proposal
In `node_modules/@expensify/react-native-web/dist/exports/UIManager` modify both `index.js` and `index.js.flow` by making this change:
```diff
@@ -00,0 +00,0 @@ var focusableElements = {
   A: true,
   INPUT: true,
   SELECT: true,
-  TEXTAREA: true
+  TEXTAREA: true,
+  BODY: true // Adding tabIndex to BODY messes up modal accessibility. Browsers do focus the body without the need of tabIndex
 };
 var UIManager = {
   blur(node) {
```
_PS: not a valid `diff`, you need to make the changes manually and not using `git apply`_

### RCA
Getting to the RCA is pretty easy as you can see we can select just fine at first and the issue only happens **after closing the modal**.
That's because at first you can see that the body tag is something like that `<body style="background-color: rgb(6, 27, 9);">` However after you close the modal it will become `<body style="background-color: rgb(6, 27, 9);" tabindex="-1">` We can see that this is related to `tabIndex` and that's why the proposals above that are focus related seems to work.
The proper fix is to understand why `tabindex="-1"` is being added and is it necessary?

Okay, so there is a known behaviour on RN modals that once you close the modal, it will reset focus to the previously focused element.

https://github.com/Expensify/react-native-web/blob/bd1f7b887ad690ef926f0c3d9f97c1f625d4bd55/packages/react-native-web/src/exports/Modal/ModalFocusTrap.js#L142-L156

and in our case the previously focused element is `<body>`.

In RNW the `focus` functionality is handled by `UIManager` in which we have a function that takes a `node` and focus it, and if that node is not focusable it adds `tabIndex="-1"` to make it focusable.

https://github.com/Expensify/react-native-web/blob/bd1f7b887ad690ef926f0c3d9f97c1f625d4bd55/packages/react-native-web/src/exports/UIManager/index.js#L50-L64

The thing is `<body>` is a special case where it can be focused without the need of `tabIndex`. You can verify that by `document.body.focus()` followed by `document.activeElement` you will get `<body>`.

Now we know why `tabIndex` is being added and we know that it is unnecessary to add and it's the root cause of this issue (and possibly others) so we can fix the issue by simply excluding `<body>` from getting that attribute.

### Results

https://user-images.githubusercontent.com/16493223/212070993-b5ee0e50-a229-4262-bf05-82ff937b2e61.mp4



___

The proposal makes changes to `node_modules` and those are not handled by `webpack` so after you apply the patch you need to clear the cache and rebuild: `rm -rf node_modules/.cache && npm run web`

I think we should raise a PR upstream and in our fork instead of applying a patch.

**Comment by joekaufmanexpensify** [comment]
Reviews of solutions still pending.

**Comment by pecanoro** [comment]
Is there a case in which we wouldn't want the text to be focusable?

**Comment by s77rt** [comment]
@pecanoro @aimane-chnaif Would you please take a look at my proposal when you get a chance?
I think we are looking to fix the issue upstream.

**Comment by getusha** [comment]
@aimane-chnaif there is no case i found either. and i explained his concern here https://github.com/Expensify/App/issues/13968#issuecomment-1380290144

**Comment by s77rt** [comment]
@aimane-chnaif 
> I don't think previously focused element is expected to be `body` as default

Why do you think so? The body is supposed to be the default, if no element is focused then the body is set to be focused. You can test that on any website that's the expected behaviour.

**Comment by s77rt** [comment]
@aimane-chnaif If you are referring that the focused element should be the button, then yes usually it is but we blur the buttons on click and moving the focus to body.
https://github.com/Expensify/App/blob/7f276b6d02a6e596abe314ff44a6197b8f2397d4/src/components/MenuItem.js#L70-L72

Why we blur the buttons on click? This was introduced by https://github.com/Expensify/App/pull/9348 to fix https://github.com/Expensify/App/issues/9248

**Comment by s77rt** [comment]
So technically this is a regression from https://github.com/Expensify/App/pull/9348

**Comment by s77rt** [comment]
@aimane-chnaif Yes, that's both a regression from our code and hidden bug in RNW.
Fixing that on RNW will fix the issue once and for all (for modals (`<BaseModal />`) and for similar popups that we use or may use in the future)

**Comment by getusha** [comment]
after the blur we need to focus the modal that's why adding `focusable` solves all these issue.

**Comment by s77rt** [comment]
I have raised a PR on RNW https://github.com/necolas/react-native-web/pull/2468
Following our upstream fix procedure I think I should create a PR on our forked repo as well (if the solution is acceptable ofc)

**Comment by joekaufmanexpensify** [comment]
Still discussing proposals

**Comment by joekaufmanexpensify** [comment]
Same.

**Comment by joekaufmanexpensify** [comment]
@aimane-chnaif when you have a chance, could you share an update on where we are this one/whether we're close to being able to proceed with any of the proposals? TY!

**Comment by s77rt** [comment]
@aimane-chnaif Even if the PR ended up not being accepted we would still be better merging it in our repo. Some changes are not necessary to be applied upstream that's why we have forks. Applying `focusable` to `<Modal />` is only a temporarily solution as we will have remove it later once we start focusing on accessibility again.

**Comment by getusha** [comment]
@s77rt i disagree, necolas even suggests using `focusable` to fix those issues. and there are a lot of issues which are fixed

**Comment by s77rt** [comment]
@getusha What part you disagree on? If it's the accessibility issue then you can verify that by TAB navigation, you will get 1 extra focus-able element (the modal). Currently we are ignoring those issues so I'm only saying that the solution is temporarily.

**Comment by getusha** [comment]
@s77rt that's not an issue and can't concern us, also fixing it is pretty easy and straight-forward if needed.

**Comment by joekaufmanexpensify** [comment]
Still reviewing proposals here.

**Comment by s77rt** [comment]
@ntdiary Calling `Element.focus()` directly won't work on elements that are not focusable. I think that's the reason `UIManager` is used.

Also `body` is a focusable element [source](https://allyjs.io/data-tables/focusable.html#document-elements)

> If current document has no focused element, document.activeElement will return body or null

That actually backs up my proposal, if body was not focsuable, `null` would be always returned.

**Comment by getusha** [comment]
`body` should be `tabbable` not only focusable for accessibility (to close the modal on enter) that's why tabindex="0" is added to body. The body element is not tabbable by default, which we need

**Comment by s77rt** [comment]
`body` is never intended too be `tabbable` and it should not.

**Comment by getusha** [comment]
Can you focus the body element with keyboard `Tab` without `tabindex`?

**Comment by s77rt** [comment]
@ntdiary `body` being focusable is more a browser thing. I think we agree that we should not add `tabIndex` as basically your solution and mine will lead into the same final results.
But I think fixing `UIManager` focus method covers more. As I would imagine there would be cases where RNW is still calling `UIManager` on other places in which the node is `body`, so fixing this on `UIManager` would fix the issue in all the places.

**Comment by joekaufmanexpensify** [comment]
Still reviewing proposals.

**Comment by joekaufmanexpensify** [comment]
@aimane-chnaif Looks like there's been a good amount of discussion on the issue. Curious if you think we're close to being able to proceed with a proposal here?

**Comment by s77rt** [comment]
@aimane-chnaif Just keep pressin TAB when a modal is open and notice the different with and without the use of `focusable`. You will find out that the TAB cycle is 1 element longer using `focusable`.
Currently the TAB navigation is entirely broken so it does not matter for now, however we will eventually fix TAB navigation and when that time comes we will have to remove `focusable` and we would run into this issue again. While fixing this on RNW is a permanent fix.

**Comment by s77rt** [comment]
@aimane-chnaif @joekaufmanexpensify We have a process to follow in the case that the fix is upstream and we do have a fork of that repo. I believe this process is to processed with the PR in our PR and also in parallel have this PR in the upstream repo to keep the divergence as little as possible.

cc @tgolen can you confirm that?

**Comment by getusha** [comment]
>Currently the TAB navigation is entirely broken 

Yes, it's broken by your PR to fix the tooltip but i don't agree this breaking the accessibility, we are just making the modal focusable

**Comment by s77rt** [comment]
@getusha If you think any of my changes broke anything please comment on the offending PR/Issue and/or report a new issue on Slack if the regression period is already over..

**Comment by getusha** [comment]
>That means we don't have to wait for the PR in RNW.

adding BODY to focusable elements is very wrong since body is not focusable/tabbable by default. I also believe necolas would disagree with that idea since it's not the right thing.

**Comment by s77rt** [comment]
Adding `tabIndex` to body is wrong. Check the source code of RNW.

**Comment by getusha** [comment]
>I have a little different opinion. 😂
UIManager.focus is designed to try focus on any element (including elements not in the keyboard tab flow).
ModalFocusTrap and UIManager should be two independent components.

As I said, the original body element is not a focusable element.
Adding the body into focusableElements does not fit the reality.
Although it can fix this issue. But Developer will no longer be able to use UIManager.focus to focus on the body and change the document.activeElement. Of course, the library should remind developers of the risks of using it. Personally, I think if RNW support to add tabindex, it also should support to remove tabindex.

as @ntdiary tried to explain multiple times, i completely agree with that

**Comment by getusha** [comment]
<img width="941" alt="Screenshot 2023-01-25 at 4 58 47 AM" src="https://user-images.githubusercontent.com/75031127/214569523-997a8f3b-3090-4d93-b881-1b9d8505ae50.png">

as far as i know body doesn't have any special styling and every component that's included in `focusableElements` is focusable by tab and body is not

**Comment by s77rt** [comment]
`focusableElements` is just a variable name, it does not mean anything, check the source code you will get what I mean.
Would you be more convinced if I rename that variable to `excludedElements` or `excludedFromTabIndex`.

**Comment by getusha** [comment]
<img width="941" alt="Screenshot 2023-01-25 at 5 08 54 AM" src="https://user-images.githubusercontent.com/75031127/214571509-cf5ccb73-3051-478c-89c7-804607fd6a56.png">

https://github.com/necolas/react-native-web/blob/ca152037bf5d268a5235fe6b500424a51c4a1b90/packages/react-native-web/src/exports/UIManager/index.js#L53-L55

It clearly states to add elements that are focusable by keyboard, and body is not focusable by keyboard

**Comment by getusha** [comment]
And all of the include elements are `tabbable` or support focus using keyboard by default
```javascript
const focusableElements = {
  A: true,
  INPUT: true,
  SELECT: true,
  TEXTAREA: true
};
```

**Comment by s77rt** [comment]
It's an implication not an equivalence. 
tabbable elements ⇒ do not add tabIndex

If we are excluding body from getting tabIndex we are not implicating that body is tabbable.

**Comment by getusha** [comment]
Setting the tabindex of the focused element to "0" ensures that if the user tabs away from the widget and then returns, the selected item within the group retains focus. Note that updating the tabindex to "0" requires also updating the previously selected item to tabindex="-1". This technique involves programmatically moving focus in response to key events and updating the tabindex to reflect the currently focused item. To do this:

**Comment by getusha** [comment]
<img width="941" alt="Screenshot 2023-01-25 at 5 53 10 AM" src="https://user-images.githubusercontent.com/75031127/214581388-5d88ed8a-9e89-4030-8406-762e635166fd.png">

#### this is totally expected behavior in our case there a lot of issues resolved by Roving `tabindex` on RNW, and we can say that it's a feature and not a bug!

https://developer.mozilla.org/en-US/docs/Web/Accessibility/Keyboard-navigable_JavaScript_widgets

https://github.com/necolas/react-native-web/issues/1099

https://github.com/necolas/react-native-web/issues/1099#issuecomment-832756341

and it doesn't make any sense to add `body` to focusableElements on RNW. we just need to make the modal `focusable`. which will be focused on tab navigation and also when we click it

**Comment by getusha** [comment]
@aimane-chnaif body is not focusable element by default

<img width="941" alt="Screenshot 2023-01-25 at 7 34 33 AM" src="https://user-images.githubusercontent.com/75031127/214605902-32c1c496-6b9f-463e-a0de-6a18b49ab071.png">

https://bitsofco.de/how-and-when-to-use-the-tabindex-attribute/

**Comment by necolas** [comment]
I'm happy to review PRs to RNW and appreciate both bug reports and fixes coming from users. But in the last few months I've receiving a few PRs from people responding to Expensify bug bounties, and I'm not enjoying getting pinged on PRs with comments like "please look at this", especially if a PR doesn't have unit tests or code comments. Please make sure people creating upstream PRs are reading the RNW contribution guidelines and following good engineering practices. This will mean that PRs can be reviewed and merged faster, and that I'm not doing work to fix PRs that other people are getting paid for. Thanks

**Comment by s77rt** [comment]
@necolas Thanks a lot for helping out. Sorry for any trouble that this may have caused you. Will make sure to have better PRs in the future!

**Comment by s77rt** [comment]
@aimane-chnaif @joekaufmanexpensify Now the PR is merged and the fix has landed upstream, I think we should move forward and apply the same fix in our forked repo.

**Comment by joekaufmanexpensify** [comment]
Still discussing in Slack.

**Comment by joekaufmanexpensify** [comment]
Same.

**Comment by joekaufmanexpensify** [comment]
Should have an update here on next steps soon.

**Comment by joekaufmanexpensify** [comment]
Okay, quick update here, we're ready to proceed. @s77rt, we're going to hire you for the upstream fix for the issue. However, at $2,000. Specifically, because the price has continued incrementing here while we've been waiting for the upstream fix.

**Comment by joekaufmanexpensify** [comment]
We need to hire: 

- @s77rt - $2,000 for upstream fix. 
- @aimane-chnaif - $2,000 for C+. 
- @gadhiyamanan - $250 for reporting the bug.

**Comment by joekaufmanexpensify** [comment]
Sounds good! I'll wait to extend the offer/assign them until we're unblocked on proceeding here.

**Comment by joekaufmanexpensify** [comment]
Actually, will hold off on hiring anyone else until the RNW repo stuff is complete.

**Comment by s77rt** [comment]
@joekaufmanexpensify Thanks for the update!
About RNW, shouldn't I raise the PR? Or this is has to be done internally due to how RNW release process work? 

cc @aimane-chnaif

**Comment by pecanoro** [comment]
I managed to test the updated upstream repo, I am not sure if I am doing something wrong but the updated repo doesn't seem to fix this bug.

**Comment by s77rt** [comment]
@pecanoro It should work. Can you verify `node_modules/@expensify/react-native-web/dist/exports/UIManager/index.js` and confirm that it contains `BODY: true`

**Comment by pecanoro** [comment]
So if I add `BODY: true`, it works, but when trying to use the changes from [here](https://github.com/Expensify/react-native-web/pull/14/files#diff-5b6ef482e15b0cf7ffb3085b2443c48fc22241dc83142b4a024f424d03722452), it doesn't seem to get fixed.

**Comment by pecanoro** [comment]
Wait, I made a stupid mistake, I kept replacing the files in `node_modules/react-native-web/src/exports/UIManager/index.js` instead of `node_modules/@expensify/react-native-web/dist/exports/UIManager/index.js`. It works well!

**Comment by pecanoro** [comment]
I am going to assign you to the issue and once I publish the changes in npm, you can create a PR to update the package.

**Comment by Beamanator** [comment]
Fork PR merged, but still need to publish the changes in npm & test here so reopened 😅

**Comment by pecanoro** [comment]
@s77rt New [version](https://www.npmjs.com/package/@expensify/react-native-web?activeTab=versions) (0.18.12) has been published to npm, feel free to create a PR now to test the changes and make sure it fixes the bug and doesn't create new ones.

**Comment by s77rt** [comment]
Applied in [Upwork](https://www.upwork.com/jobs/~0142c09828b956ad3b).
[PR](https://github.com/Expensify/App/pull/14870) is ready.

**Comment by joekaufmanexpensify** [comment]
>Applied in [Upwork](https://www.upwork.com/jobs/~0142c09828b956ad3b).

Great, thanks! Offer sent.

**Comment by joekaufmanexpensify** [comment]
@gadhiyamanan Offer sent for $250!

**Comment by MelvinBot** [comment]
`Reviewing` label has been removed, please complete the "BugZero Checklist".

**Comment by MelvinBot** [comment]
The solution for this issue has been :rocket: deployed to production :rocket: in version 1.2.67-7 and is now subject to a 7-day regression period :calendar:. Here is the list of pull requests that resolve this issue:

- https://github.com/Expensify/App/pull/14870


If no regressions arise, payment will be issued on 2023-02-15. :confetti_ball:

After the hold period is over and BZ checklist items are completed, please complete any of the applicable payments for this issue, and check them off once done.
- [ ] External issue reporter
- [ ] Contributor that fixed the issue
- [ ] Contributor+ that helped on the issue and/or PR

As a reminder, here are the bonuses/penalties that should be applied for any External issue:
- Merged PR within 3 business days of assignment - 50% **bonus**
- Merged PR more than 9 business days after assignment - 50% **penalty**

**Comment by MelvinBot** [comment]
**BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

- [ ] [@aimane-chnaif / @pecanoro] The PR that introduced the bug has been identified. Link to the PR:
- [ ] [@aimane-chnaif / @pecanoro] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:
- [ ] [@aimane-chnaif / @pecanoro] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:
- [ ] [@s77rt] Propose [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the same bug will not reach production again.
- [ ] [@joekaufmanexpensify] Link the GH issue for creating/updating the regression test once above steps have been agreed upon:

**Comment by s77rt** [comment]
**Regression Test Proposal**
1. Go to Settings > About
2. Open "View Keyboard shortcuts" modal
3. Verify that you can select text
4. Close modal
5. Reopen same modal
6. Verify that you can still select text

**Comment by joekaufmanexpensify** [comment]
We're going to hit the 7 day regression waiting period later today, so we'll be all set to issue payment here soon. In the meantime, the above regression test steps look good to me. I'll work on creating an issue to add this new regression test. 

Besides that, all that's left for the BZ checklist are @pecanoro / @aimane-chnaif taking the steps to identify and comment on the offending PR!

**Comment by joekaufmanexpensify** [comment]
Okay, added regression test above! All that's left before we can issue payment here is for @aimane-chnaif / @pecanoro to have the discussion in #expensify-bugs about whether there are any changes we should make to the review checklist, etc. to  better prevent this type of bug going forward!

**Comment by joekaufmanexpensify** [comment]
Great, thanks! BZ checklist is complete. Issuing payment now!

**Comment by joekaufmanexpensify** [comment]
@s77rt $2000 paid, and contract ended!

**Comment by joekaufmanexpensify** [comment]
@aimane-chnaif $2000 paid, and contract ended!

**Comment by joekaufmanexpensify** [comment]
@gadhiyamanan $250 paid, and contract ended!

**Comment by joekaufmanexpensify** [comment]
Upwork job closed.

**Comment by joekaufmanexpensify** [comment]
Okay, bug is fixed, BZ checklist is complete, and all payments are issued. This is all set. Thanks everyone!

**Comment by s77rt** [comment]
Thanks! @joekaufmanexpensify 
Just for confirmation, doesn't bonus timeline apply here? I have been hold assigned for RNW update.
